plugins/jobtype/src/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -316,6 +319,41 @@ public boolean accept(File dir, String name) {
     return matchingFiles;
   }
 
+/**
+ * This method is a decorator around the KillAllSpawnedHadoopJobs method.
+ * This method takes additional parameters to determine whether KillAllSpawnedHadoopJobs needs to be executed 
+ * using doAs as a different user
+ * 
+ * @param logFilePath Azkaban log file path
+ * @param jobProps Azkaban job props
+ * @param tokenFile Pass in the tokenFile if value is known.  It is ok to skip if the token file is in the environmental variable
+ * @param log a usable logger
+ */
+  public static void proxyUserKillAllSpawnedHadoopJobs(final String logFilePath, Props jobProps, File tokenFile, final Logger log) {
```

Comment:
as we talked offline once we have the security feature permanently in we may no longer need to proxy user when killing the jobs as the currently running credential is the submitting/proxy to user anyways,  I agree it is ok to have this logic in right now as the security feature is not in yet , it will be nice to create a followup task to refector this piece of once when the feature is in.